applications/GeoMechanicsApplication/custom_conditions/T_normal_flux_condition.h

```diff
@@ -0,0 +1,83 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#pragma once
+
+#include "custom_conditions/T_condition.h"
+#include "custom_utilities/condition_utilities.hpp"
+#include "custom_utilities/element_utilities.hpp"
+#include "geo_mechanics_application_variables.h"
+#include "includes/condition.h"
+#include "includes/serializer.h"
+
+namespace Kratos {
+
+template <unsigned int TDim, unsigned int TNumNodes>
+class KRATOS_API(GEO_MECHANICS_APPLICATION) TNormalFluxCondition
+    : public TCondition<TDim, TNumNodes> {
+public:
+    using NodeType = Node;
+    using GeometryType = Geometry<NodeType>;
+    using PropertiesType = Properties;
+    using NodesArrayType = GeometryType::PointsArrayType;
+    using VectorType = Vector;
+
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(TNormalFluxCondition);
+
+    TNormalFluxCondition();
+
+    TNormalFluxCondition(IndexType NewId, GeometryType::Pointer pGeometry);
+
+    TNormalFluxCondition(IndexType NewId,
+                         GeometryType::Pointer pGeometry,
+                         PropertiesType::Pointer pProperties);
+
+    ~TNormalFluxCondition() override;
+
+    Condition::Pointer Create(IndexType NewId,
+                              NodesArrayType const& rThisNodes,
+                              PropertiesType::Pointer pProperties) const override;
+
+protected:
+    struct NormalFluxVariables {
+        double normalFlux;
+        double IntegrationCoefficient;
+        array_1d<double, TNumNodes> N;
+        array_1d<double, TNumNodes> fluxVector;
+    };
```

Comment:
Could you make sure these are consistent (so NormalFlux, FluxVector) and maybe give a more descriptive name for N? Is it the normal vector?

Reply:
NormalFlux/FluxVector are used now. For N, I had a discussion with Mohamed. That is difficult.